Test test_resample_loffset fails on i586 with Python 3.9

<!-- Please include a self-contained copy-pastable example that generates the issue if possible.

Please be concise with code posted. See guidelines below on how to provide a good bug report:

- Craft Minimal Bug Reports: http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports
- Minimal Complete Verifiable Examples: https://stackoverflow.com/help/mcve

Bug reports that follow these guidelines are easier to diagnose, and so are often handled much more quickly.
-->

**What happened**:
Running tests with `pytest-3.9 -v -n auto xarray`
 ->
```
[  262s] =================================== FAILURES ===================================
[  262s] ______________________ TestDataset.test_resample_loffset _______________________
[  262s] [gw1] linux -- Python 3.9.5 /usr/bin/python3.9
[  262s] 
[  262s] self = <xarray.tests.test_dataset.TestDataset object at 0xf0719a18>
[  262s] 
[  262s]     def test_resample_loffset(self):
[  262s]         times = pd.date_range("2000-01-01", freq="6H", periods=10)
[  262s]         ds = Dataset(
[  262s]             {
[  262s]                 "foo": (["time", "x", "y"], np.random.randn(10, 5, 3)),
[  262s]                 "bar": ("time", np.random.randn(10), {"meta": "data"}),
[  262s]                 "time": times,
[  262s]             }
[  262s]         )
[  262s]         ds.attrs["dsmeta"] = "dsdata"
[  262s]     
[  262s]         # Our use of `loffset` may change if we align our API with pandas' changes.
[  262s]         # ref https://github.com/pydata/xarray/pull/4537
[  262s]         actual = ds.resample(time="24H", loffset="-12H").mean().bar
[  262s]         expected_ = ds.bar.to_series().resample("24H").mean()
[  262s]         expected_.index += to_offset("-12H")
[  262s]         expected = DataArray.from_series(expected_)
[  262s] >       assert_identical(actual, expected)
[  262s] E       AssertionError: Left and right DataArray objects are not identical
[  262s] E       
[  262s] E       Differing values:
[  262s] E       L
[  262s] E           array([ 0.018477, -0.178807, -0.792644])
[  262s] E       R
[  262s] E           array([ 0.018477, -0.178807, -0.792644])
[  262s] 
[  262s] /home/abuild/rpmbuild/BUILD/xarray-0.18.1/xarray/tests/test_dataset.py:3991: AssertionError
```
[devel:languages:python:numeric-python-xarray-openSUSE_Tumbleweed-i586.txt](Complete log)

**What you expected to happen**:
Tests succeed

**Minimal Complete Verifiable Example**:

```python
# Put your MCVE code here
```

**Anything else we need to know?**:
Tests with x86_64 succeed

**Environment**:

openSUSE Tumbleweed build environment (for packaging)

<details><summary>Output of <tt>xr.show_versions()</tt></summary>

```

INSTALLED VERSIONS
------------------
commit: None
python: 3.9.5 (default, May 05 2021, 15:23:13) [GCC]
python-bits: 32
OS: Linux
OS-release: 5.12.3-1-default
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: None
LOCALE: ('en_US', 'UTF-8')
libhdf5: None
libnetcdf: None

xarray: 0.18.1
pandas: 1.2.4
numpy: 1.20.2
scipy: 1.6.3
netCDF4: None
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: None
cftime: None
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: None
dask: 2021.04.1
distributed: None
matplotlib: None
cartopy: None
seaborn: None
numbagg: None
pint: None
setuptools: 44.1.1
pip: None
conda: None
pytest: 6.2.2
IPython: None
sphinx: None
```

</details>

--- a/xarray/tests/test_dataset.py
+++ b/xarray/tests/test_dataset.py
@@ -3987,7 +3987,7 @@ class TestDataset:
         expected_ = ds.bar.to_series().resample("24H").mean()
         expected_.index += to_offset("-12H")
         expected = DataArray.from_series(expected_)
-        assert_identical(actual, expected)
+        assert_allclose(actual, expected)
 
     def test_resample_by_mean_discarding_attrs(self):
         times = pd.date_range("2000-01-01", freq="6H", periods=10)

